presto-main/src/main/java/com/facebook/presto/server/protocol/QueryResourceUtil.java

```diff
@@ -274,4 +290,70 @@ private static int countStageAndAddGlobalUniqueNodes(List<TaskInfo> tasks, Set<S
         }
         return stageUniqueNodes.size();
     }
+
+    /**
+     * We follow the type signatures to track the data type and serializing nested data structure.
```

Comment:
Can you add some more detail on the problem--that we'll attempt to render this as JSON, but JSON only supports string keys?

Reply:
I modified the comment, added more detail and the description of the problem.